pkg/kubernetes/reconcile.go

```diff
@@ -20,14 +20,37 @@ func (e ErrorPrimitiveReached) Error() string {
 }
 
 // walkJSON traverses deeply nested kubernetes manifest and extracts them into a flat []dict.
-func walkJSON(deep map[string]interface{}, path string) ([]map[string]interface{}, error) {
+func walkJSON(rawDeep interface{}, path string) ([]map[string]interface{}, error) {
+	flat := make([]map[string]interface{}, 0)
+
+	// array: walkJSON for each
+	if d, ok := rawDeep.([]map[string]interface{}); ok {
```

Comment:
Just want to point out that this wouldn't support nested arrays, which might not be something we want t support right now anyway.

Reply:
I'm fine with this limitation, as the implementation would become much more complex, due to extensive type checking. 

Would like to encourage users to use dicts anyways